analysis/deciles_charts.py

```diff
@@ -81,17 +102,24 @@ def parse_args():
         type=get_path,
         help="Path to the output directory",
     )
+    parser.add_argument(
+        "--config",
+        default=DEFAULT_CONFIG.copy(),
+        type=parse_config,
```

Comment:
This was confusing to me initially as it's not really type checking. Is it common do to error handling like this with argparse?

Reply:
Another great question, and the answer is "it depends". The functions we pass to `type` do two things: they coerce strings to other types; and they handle errors. The former is [a documented use of `type`][1]. The latter can be debated. On the one hand, type coercion will require some error handling; on the other hand, handling invalid keys is different to handling invalid JSON.

My (current!) preference it to put type coercion and error handling in the functions we pass to `type`, and accept that `type` is a poor choice of name. Doing so means that `args` contains valid, typed data; it also feels consistent with [Click][2], which people who are far more intelligent than me have spent a lot of time thinking about 🙂 

[1]: https://docs.python.org/3.8/library/argparse.html#type
[2]: https://click.palletsprojects.com/en/8.1.x/

Reply:
I think it's neat way of doing it! Will check out Click

Reply:
Click isn't a direct dependency of python-docker, so I've not used it for reusable actions. It is a transitive dependency (a dependency of a dependency), but I don't think it's a good idea to depend upon transitive dependencies.